### PR TITLE
Add role Test case

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #1063
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/tests/test_filters.py [hunk 1]
```diff
@@ -1516,3 +1518,52 @@ def test_search(self):
...
+            Role.objects.create(name="Role 1", color=ColorChoices.COLOR_GREY, weight=100),
+            Role.objects.create(name="Role 2", color=ColorChoices.COLOR_GREY, weight=20),
+            Role.objects.create(name="Role 3", color=ColorChoices.COLOR_AMBER, weight=10),
+            Role.objects.create(name="Role 4", color=ColorChoices.COLOR_AMBER, weight=100),
+            Role.objects.create(name="Role 5", color=ColorChoices.COLOR_AMBER, weight=100),
+        ]
+        for i in range(2):
+            roles[i].content_types.add(cls.device_ct)
+        for i in range(2, 4):
+            roles[i].content_types.add(cls.rack_ct)
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_filters.py:1541]: Can we consider instead adding a RoleFactory?

File: nautobot/extras/tests/test_filters.py [hunk 2]
```diff
@@ -1516,3 +1518,52 @@ def test_search(self):
...
+
+    def test_content_types(self):
+        device_count = self.queryset.filter(content_types=self.device_ct).count()
+        params = {"content_types": ["dcim.device"]}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), device_count)
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_filters.py:1546]: These should be using `assertQuerysetEqualAndNotEmpty`.

File: nautobot/extras/urls.py [hunk 3]
```diff
@@ -552,6 +553,19 @@
...
+        views.ObjectChangeLogView.as_view(),
+        name="role_changelog",
+        kwargs={"model": Role},
+    ),
+    path(
+        "roles/<str:slug>/notes/",
+        views.ObjectNotesView.as_view(),
+        name="role_notes",
+        kwargs={"model": Role},
+    ),
```
Comment (glennmatthews) [inline nautobot/extras/urls.py:568]: These will go away once converting the Role views to a NautobotUIViewSet, right?

Comment (HanlinMiao) [reply nautobot/extras/urls.py:568]: I think `Role` views are already using `NautobotUIViewSet`. But `ChangeLogMixin` and `NotesMixin` are not in the ViewSet on this branch for some reason. Maybe this branch is not up to date with the base branch?

File: nautobot/utilities/tests/test_templatetags_helpers.py [hunk 4]
```diff
@@ -209,3 +214,15 @@ def test_render_boolean(self):
                 render_boolean(value), '<span class="text-danger"><i class="mdi mdi-close-thick" title="No"></i></span>'
             )
         self.assertEqual(render_boolean(None), '<span class="text-muted">&mdash;</span>')
+
+    def test_render_obj_display(self):
+        vlan_ct = ContentType.objects.get_for_model(VLAN)
+        role = Role.objects.create(name="New Role", color=ColorChoices.COLOR_GREY)
+        role.content_types.add(vlan_ct)
+        site = Site.objects.create(name="Site-1", slug="site-1")
+        vlan = VLAN.objects.create(name="VLAN 1", vid=100, site=site, role=role)
```
Comment (HanlinMiao) [inline nautobot/utilities/tests/test_templatetags_helpers.py:223]: Can we use the factory test fixture here instead?

File: nautobot/extras/urls.py (hunk 3 above)
Comment (timizuoebideri1) [reply nautobot/extras/urls.py:568]: Nope, NautobotUIViewSet do not account for object_change_log and notes

Comment (smk4664) [reply nautobot/extras/urls.py:568]: It should, I verified that was working with the CircuitsViewSet

Comment (smk4664) [reply nautobot/extras/urls.py:568]: Oh, looks like the Role model doesn't have the notesmixin, as far as I can see.

Comment (smk4664) [reply nautobot/extras/urls.py:568]: The NautobotUIViewSetRouter should have the Urls in it.

Comment (timizuoebideri1) [reply nautobot/extras/urls.py:568]: Thats for api, This is UI

Comment (timizuoebideri1) [reply nautobot/extras/urls.py:568]: nautobot.circuits.urls.py:L55
```python
path(
        "circuits/<uuid:pk>/changelog/",
        ObjectChangeLogView.as_view(),
        name="circuit_changelog",
        kwargs={"model": Circuit},
    ),
    path(
        "circuits/<uuid:pk>/notes/",
        ObjectNotesView.as_view(),
        name="circuit_notes",
        kwargs={"model": Circuit},
    ),
```

Comment (smk4664) [reply nautobot/extras/urls.py:568]: Weird, my PR has it removed. I will send it when I find the PR on the computer, can't link the lines from my phone. This is most definitely not just the API, and there might have been a regression.

Comment (smk4664) [reply nautobot/extras/urls.py:568]: In addition, these are the old views, not the ViewSets.

Comment (smk4664) [reply nautobot/extras/urls.py:568]: Check out the develop branch, it does not have those urls.

Comment (timizuoebideri1) [reply nautobot/extras/urls.py:568]: oooh True, Illd have to update my branch. Thanks

**Pull request merged**